File: algorithm/sliding-window/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        res = 0
        tmp = ""
        for i in s:
            if i not in tmp:
                tmp += i
            else:
                tmp = tmp[tmp.index(i) + 1:] + i
                # print(tmp)
            if len(tmp) > res:
                res = len(tmp)
        return res

File: algorithm/sliding-window/3.go
package main

import (
	"fmt"
	"slices"
)

func main() {
	fmt.Println(lengthOfLongestSubstring("abcabcbb")) // 3
}

func lengthOfLongestSubstring(s string) int {
	t := []rune("")
	res := 0
	for i := range s {
		// fmt.Println(string(t), " ", i)
		if !slices.Contains(t, rune(s[i])) {
			t = append(t, rune(s[i]))
		} else {
			index := slices.Index(t, rune(s[i]))
			t = append(t[index+1:], rune(s[i]))
		}

		if len(t) > res {
			res = len(t)
		}
	}
	return res
}
